Feature/sc 1 be create blank project in asp net core

lgaljer20 opened the pull request:
Created 3-tier layer architecture that consists of: API, Business and Data Layer.
Installed NuGet packages that are responsible for communication between PostgreSQL and Entity Framework.
In Data layer added database context  and made entities for database.
Added connection string and made initial migration.

lgaljer20 committed:
Installed NuGet packages for PostgreSQL and Entity Framework

lgaljer20 committed:
Added business and data layers to project and their project references.

lgaljer20 committed:
Added empty DTO, Interface and Service folders inside business logic layer.

lgaljer20 committed:
Added NuGet packages in data layer

lgaljer20 committed:
Added Entity Framework Core Context and Entity Sets.
Defined entity sets for each data entity (User, Role, Charger, Event, Card) that will be mapped to corresponding database tables.
Added two types of role in context.

lgaljer20 committed:
Configured database connection

lgaljer20 committed:
Made initial migration to the database.

lraknic reviewed (approved):
Approved, everything looks good.

lgaljer20 committed:
Made LastSync and EndTime and Volume nullable
In Charger entity LastSync can be nullable if it's the first event and in Event entity before stopping charging EndTime and Volume are always nullable
Made new migration to fix the database

lraknic20 commented:
Thanks for addressing this oversight. I have reviewed the changes and they look good to me.

mkajic20 commented on SmartCharger.Data/Entities/Charger.cs:
```diff
@@ -0,0 +1,20 @@
+        public string Name { get; set; }
+        public double Latitude { get; set; }
+        public double Longitude { get; set; }
+        public  DateTime Created { get; set; }
```
I suggest a different name for this variable so the name specifies that it is referring to time when it was created. (for example, CreationTime)

lgaljer20 replied:
I agree, thank you.  In new commit I will change the name of the variable.

mkajic20 commented on SmartCharger.Data/Entities/Event.cs:
```diff
@@ -0,0 +1,23 @@
+        public int ChargerId { get; set; }
+        public virtual Charger Charger { get; set; }
+
+        [ForeignKey(nameof(Card))]
```
Since Card has relation with User where it can only be linked to one user, why did you create a foreign key for both card and user?

lgaljer20 replied:
I gave this a lot of thought and tried to find edge cases. I've found a case where some user may delete his card and then give that same physical card to a different user. In that case, another user would be able to see his events (history of charging).

mkajic20 replied:
That makes sense, thank you for the explanation. In that case there is no need for changes in that file.

mkajic20 commented on SmartCharger.Data/Entities/User.cs:
```diff
@@ -0,0 +1,20 @@
+        public string Email { get; set; }
+        public string Password { get; set; }
+        public bool? Active { get; set; }
+        public DateTime Created { get; set; }
```
Same comment as with Charger class, I think this variable should be named differently.

mkajic20 reviewed (changes requested):
Looks good, I left a few comments for potential changes.

lgaljer20 committed:
New changes in naming the variables based on PR.

lgaljer20 commented:
@mkajic20 I've changed the names of the variables and wrote an explanation. Hope everything is clear now. If everything looks correct, please merge and close this pull request.

mkajic20 reviewed (approved):
Everything looks great now, thank you for accepting my feedback.
